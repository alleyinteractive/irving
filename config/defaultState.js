export const components = [];
export const route = {
  notFound: false,
  action: '',
  location: {},
  siteComponents: [],
  pages: {},
};

const defaultState = {
  components,
  route,
};

export default defaultState;
