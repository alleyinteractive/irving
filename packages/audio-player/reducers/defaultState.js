const getDefaultState = () => ({
  player: {
    currentTime: 0,
    duration: 0,
    loading: false,
    playing: false,
    seek: 0,
    src: '',
    visible: false,
    volume: 0.5,
  },
});

export default getDefaultState;
