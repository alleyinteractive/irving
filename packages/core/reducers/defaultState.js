import { getValueFromMergedConfig } from 'config/irving/getValueFromMergedConfig';

export const components = {
  defaults: [],
  providers: {},
  page: {},
};

export const route = {
  status: null,
  redirectTo: false,
  redirectStatus: 0,
  action: '',
};

export const visible = {
  test: false,
};

export const error = null;

export const loading = false;

export const componentData = {};

export const componentDataMeta = {
  loading: false,
  loaded: false,
  error: false,
  data: [],
};

const defaultState = {
  components,
  componentData,
  error,
  loading,
  route,
  visible,
};

export default getValueFromMergedConfig('defaultState', defaultState);
