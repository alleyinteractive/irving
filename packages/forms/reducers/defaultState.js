export const formState = {
  submitting: false,
  submitted: false,
  failed: false,
  validation: {},
  redirect: '',
};

export default {
  forms: {},
};
