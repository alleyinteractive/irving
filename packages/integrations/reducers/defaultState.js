export const manager = {
  componentMap: [],
  hydrated: false,
};

export const pico = {};

export const coral = {};

export default {
  manager,
  pico,
  coral,
};
