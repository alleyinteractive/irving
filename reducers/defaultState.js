export const components = {
  defaults: [],
  providers: {},
  page: {},
};

export const route = {
  status: null,
  redirectTo: false,
  redirectStatus: 0,
  action: '',
};

export const form = {
  submitting: false,
  submitted: false,
  failed: false,
  validation: {},
  redirect: '',
};

export const visible = {
  megaMenu: false,
};

export const player = {
  currentTime: 0,
  duration: 0,
  loading: false,
  playing: false,
  seek: 0,
  src: '',
  visible: false,
  volume: 0.5,
};

export const story = {
  showFullText: false,
};

export const error = null;

export const loading = false;

export const componentData = {};

export const componentDataMeta = {
  loading: false,
  error: false,
  data: [],
};

export const headerHeight = 0;

const defaultState = {
  components,
  componentData,
  error,
  headerHeight,
  loading,
  player,
  route,
  visible,
};

export const zephr = {
  isLoading: true,
  forms: {},
  cached: false,
  session: {},
  user: {
    account: {},
    profile: {},
  },
};

export const zephrRules = {
  isLoading: false,
  components: {},
};

export default defaultState;
