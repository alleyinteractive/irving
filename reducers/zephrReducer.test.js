/* eslint-disable max-len */
import { merge } from 'lodash/fp';
import {
  actionReceiveLoginError,
  actionReceiveRegistrationError,
  actionReceiveInvalidPassword,
} from 'actions/zephrActions';
import defaultState from './defaultState';
import reducer, {
  setFormErrorState,
  setPasswordErrorState,
} from './zephrReducer';

// Default login form state to be modified in the test suite.
const loginFormMock = {
  components: JSON.stringify([
    {
      autoComplete: 'username',
      className: 'zephr-input-email-address',
      defaultValue: '',
      id: 'email-address',
      placeholder: 'Enter your email address',
      required: false,
      type: 'email',
    },
    {
      autoComplete: 'current-password',
      className: 'zephr-input-current-password',
      defaultValue: '',
      id: 'current-password',
      placeholder: 'Enter your password',
      required: true,
      type: 'password',
    },
    {
      id: 'submit-button',
      key: 'zephr-submit-button',
      type: 'submit',
      value: 'Login',
    }
  ]),
  error: false,
  errors: [],
  errorCount: null,
};

// Default registration form state to be modified in the test suite.
const regFormMock = {
  components: JSON.stringify([
    {
      autoComplete: 'username',
      className: 'zephr-input-email-address',
      defaultValue: '',
      id: 'email-address',
      placeholder: 'Enter your email address',
      required: false,
      type: 'email',
    },
    {
      className: 'zephr-input-full-name',
      defaultValue: '',
      id: 'full-name',
      placeholder: 'Enter your full name',
      required: true,
      type: 'text',
    },
    {
      autoComplete: 'new-password',
      className: 'zephr-input-new-password',
      defaultValue: '',
      id: 'new-password',
      placeholder: 'Create a password for your account',
      required: true,
      type: 'password',
    },
    {
      autoComplete: 'new-password',
      className: 'zephr-input-verify-password',
      defaultValue: '',
      id: 'verify-password',
      placeholder: 'Confirm your password',
      required: true,
      type: 'password',
    },
    {
      id: 'terms-checkbox',
    },
    {
      id: 'submit-button',
      key: 'zephr-submit-button',
      type: 'submit',
      value: 'Create an account',
    },
  ]),
  error: false,
  errors: [],
  errorCount: null,
};

/**
 * A helper function that takes a given form type, the clean form's state,
 * the target error, and the mocked state and tests whether or not the error
 * applied by the `setFormErrorState` function produces the same results as
 * the effects of a dispatched error action.
 *
 * @param {*} type      The form type.
 * @param {*} formState The (clean) form's state.
 * @param {*} error     The error to be tested.
 * @param {*} mockState The mocked state object.
 */
function compareFormState(type, formState, error, mockState) {
  const components = setFormErrorState(formState, error);

  // Run the action and return the modified state.
  let nextState = {};
  if ('login' === type) {
    nextState = reducer(mockState, actionReceiveLoginError(error))

    expect(nextState).toEqual({
      ...mockState,
      forms: {
        ...mockState.forms,
        [type]: {
          components,
          error: true,
          errors: [error],
          errorCount: 1,
          requireCaptcha: false,
        },
      },
    });
  } else if ('register' === type) {
    nextState = reducer(mockState, actionReceiveRegistrationError(error))

    expect(nextState).toEqual({
      ...mockState,
      forms: {
        ...mockState.forms,
        [type]: {
          components,
          error: true,
          errors: [error],
        },
      },
    });
  }
}

describe('Zephr Reducer', () => {
  describe('Login errors', () => {
    it('Should invalidate the email address field when an incorrect email is entered', () => {
      const mockState = merge(defaultState, {
        forms: {
          login: loginFormMock,
        },
      });
      const error = 'user-not-found';
      // Compare the state of manually errored form components and those formatted by the action.
      compareFormState('login', loginFormMock, error, mockState);

      // Create/compare a snapshot.
      const nextState = reducer(mockState, actionReceiveLoginError(error));
      expect(nextState).toMatchSnapshot();
    });

    it('Should invalidate the email address field when an unverified email is entered', () => {
      const mockState = merge(defaultState, {
        forms: {
          login: loginFormMock,
        },
      });
      const error = 'email-not-verified';
      // Compare the state of manually errored form components and those formatted by the action.
      compareFormState('login', loginFormMock, error, mockState);

      // Create/compare a snapshot.
      const nextState = reducer(mockState, actionReceiveLoginError(error));
      expect(nextState).toMatchSnapshot();
    });

    it('Should invalidate the password field when an incorrect password is entered', () => {
      const mockState = merge(defaultState, {
        forms: {
          login: loginFormMock,
        },
      });
      const error = 'invalid-password';
      // Compare the state of manually errored form components and those formatted by the action.
      compareFormState('login', loginFormMock, error, mockState);

      // Create/compare a snapshot.
      const nextState = reducer(mockState, actionReceiveLoginError(error));
      expect(nextState).toMatchSnapshot();
    });

    it('Should require a reCAPTCHA field to be filled out if a failed attempt is made 3 or more times', () => {
      const mockState = merge(defaultState, {
        forms: {
          login: {
            ...loginFormMock,
            error: true,
            errorCount: 2,
            requireCaptcha: false,
          },
        },
      });
      const error = 'user-not-found';
      const components = setFormErrorState(loginFormMock, error);

      // Run the action and return the modified state.
      const nextState = reducer(mockState, actionReceiveLoginError(error));
      expect(nextState).toEqual({
        ...mockState,
        forms: {
          ...mockState.forms,
          login: {
            components,
            error: true,
            errors: [error],
            errorCount: 3,
            requireCaptcha: true,
          },
        },
      });
    });
  });

  describe('Registration errors', () => {
    it('Should invalidate the email address field when an invalid email is entered', () => {
      const mockState = merge(defaultState, {
        forms: {
          register: regFormMock,
        },
      });
      const error = 'email-address';
      // Compare the state of manually errored form components and those formatted by the action.
      compareFormState('register', regFormMock, error, mockState);

      // Create/compare a snapshot.
      const nextState = reducer(mockState, actionReceiveRegistrationError(error));
      expect(nextState).toMatchSnapshot();
    });

    it('Should invalidate the email address field when an existing account email is entered', () => {
      const mockState = merge(defaultState, {
        forms: {
          register: regFormMock,
        },
      });
      const error = 'user-already-exists';
      // Compare the state of manually errored form components and those formatted by the action.
      compareFormState('register', regFormMock, error, mockState);

      // Create/compare a snapshot.
      const nextState = reducer(mockState, actionReceiveRegistrationError(error));
      expect(nextState).toMatchSnapshot();
    });

    it('Should invalidate the full name field when an invalid name is entered', () => {
      const mockState = merge(defaultState, {
        forms: {
          register: regFormMock,
        },
      });
      const error = 'full-name';
      // Compare the state of manually errored form components and those formatted by the action.
      compareFormState('register', regFormMock, error, mockState);

      // Create/compare a snapshot.
      const nextState = reducer(mockState, actionReceiveRegistrationError(error));
      expect(nextState).toMatchSnapshot();
    });

    it('Should invalidate the password field when a weak password is entered', () => {
      const mockState = merge(defaultState, {
        forms: {
          register: regFormMock,
        },
      });
      const error = 'password-not-strong';
      // Compare the state of manually errored form components and those formatted by the action.
      compareFormState('register', regFormMock, error, mockState);

      // Create/compare a snapshot.
      const nextState = reducer(mockState, actionReceiveRegistrationError(error));
      expect(nextState).toMatchSnapshot();
    });

    it('Should invalidate the password and verification fields when the passwords do not match', () => {
      const mockState = merge(defaultState, {
        forms: {
          register: regFormMock,
        },
      });
      const components = setPasswordErrorState(regFormMock);

      const nextState = reducer(mockState, actionReceiveInvalidPassword())
      expect(nextState).toEqual({
        ...mockState,
        forms: {
          ...mockState.forms,
          register: {
            components,
            error: true,
            errors: ['verify-password'],
          },
        },
      });
      expect(nextState).toMatchSnapshot();
    });

    it('Should invalidate the terms of service checkbox if not checked', () => {
      const mockState = merge(defaultState, {
        forms: {
          register: regFormMock,
        },
      });
      const error = 'terms-checkbox';
      // Compare the state of manually errored form components and those formatted by the action.
      compareFormState('register', regFormMock, error, mockState);

      // Create/compare a snapshot.
      const nextState = reducer(mockState, actionReceiveRegistrationError(error));
      expect(nextState).toMatchSnapshot();
    });
  });
});
