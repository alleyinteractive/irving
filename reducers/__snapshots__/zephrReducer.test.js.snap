// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zephr Reducer Login errors Should invalidate the email address field when an incorrect email is entered 1`] = `
Object {
  "componentData": Object {},
  "components": Object {
    "defaults": Array [],
    "page": Object {},
    "providers": Object {},
  },
  "error": null,
  "forms": Object {
    "login": Object {
      "components": "[{\\"autoComplete\\":\\"username\\",\\"className\\":\\"zephr-input-email-address\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"email-address\\",\\"placeholder\\":\\"Enter your email address\\",\\"required\\":false,\\"type\\":\\"email\\",\\"invalid\\":true},{\\"type\\":\\"span\\",\\"id\\":\\"email-address-error\\",\\"key\\":\\"email-address-error-message\\",\\"className\\":\\"form-error\\",\\"message\\":\\"Oops! Let’s try that again — User not found.\\\\n       Please enter your email address.\\"},{\\"autoComplete\\":\\"current-password\\",\\"className\\":\\"zephr-input-current-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"current-password\\",\\"placeholder\\":\\"Enter your password\\",\\"required\\":true,\\"type\\":\\"password\\"},{\\"id\\":\\"submit-button\\",\\"key\\":\\"zephr-submit-button\\",\\"type\\":\\"submit\\",\\"value\\":\\"Login\\"}]",
      "error": true,
      "errorCount": 1,
      "errors": Array [
        "user-not-found",
      ],
      "requireCaptcha": false,
    },
  },
  "headerHeight": 0,
  "loading": false,
  "player": Object {
    "currentTime": 0,
    "duration": 0,
    "loading": false,
    "playing": false,
    "seek": 0,
    "src": "",
    "visible": false,
    "volume": 0.5,
  },
  "route": Object {
    "action": "",
    "redirectStatus": 0,
    "redirectTo": false,
    "status": null,
  },
  "visible": Object {
    "megaMenu": false,
    "sliderAd": false,
    "sliderAdHasClosed": false,
  },
}
`;

exports[`Zephr Reducer Login errors Should invalidate the email address field when an unverified email is entered 1`] = `
Object {
  "componentData": Object {},
  "components": Object {
    "defaults": Array [],
    "page": Object {},
    "providers": Object {},
  },
  "error": null,
  "forms": Object {
    "login": Object {
      "components": "[{\\"autoComplete\\":\\"username\\",\\"className\\":\\"zephr-input-email-address\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"email-address\\",\\"placeholder\\":\\"Enter your email address\\",\\"required\\":false,\\"type\\":\\"email\\",\\"invalid\\":true},{\\"type\\":\\"span\\",\\"id\\":\\"email-address-error\\",\\"key\\":\\"email-address-error-message\\",\\"className\\":\\"form-error\\",\\"message\\":\\"Oops! Let’s try that again — Your email has not\\\\n       yet been verified. Please check your inbox\\\\n       for a verification email and try again.\\"},{\\"autoComplete\\":\\"current-password\\",\\"className\\":\\"zephr-input-current-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"current-password\\",\\"placeholder\\":\\"Enter your password\\",\\"required\\":true,\\"type\\":\\"password\\"},{\\"id\\":\\"submit-button\\",\\"key\\":\\"zephr-submit-button\\",\\"type\\":\\"submit\\",\\"value\\":\\"Login\\"}]",
      "error": true,
      "errorCount": 1,
      "errors": Array [
        "email-not-verified",
      ],
      "requireCaptcha": false,
    },
  },
  "headerHeight": 0,
  "loading": false,
  "player": Object {
    "currentTime": 0,
    "duration": 0,
    "loading": false,
    "playing": false,
    "seek": 0,
    "src": "",
    "visible": false,
    "volume": 0.5,
  },
  "route": Object {
    "action": "",
    "redirectStatus": 0,
    "redirectTo": false,
    "status": null,
  },
  "visible": Object {
    "megaMenu": false,
    "sliderAd": false,
    "sliderAdHasClosed": false,
  },
}
`;

exports[`Zephr Reducer Login errors Should invalidate the password field when an incorrect password is entered 1`] = `
Object {
  "componentData": Object {},
  "components": Object {
    "defaults": Array [],
    "page": Object {},
    "providers": Object {},
  },
  "error": null,
  "forms": Object {
    "login": Object {
      "components": "[{\\"autoComplete\\":\\"username\\",\\"className\\":\\"zephr-input-email-address\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"email-address\\",\\"placeholder\\":\\"Enter your email address\\",\\"required\\":false,\\"type\\":\\"email\\"},{\\"autoComplete\\":\\"current-password\\",\\"className\\":\\"zephr-input-current-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"current-password\\",\\"placeholder\\":\\"Enter your password\\",\\"required\\":true,\\"type\\":\\"password\\",\\"invalid\\":true},{\\"type\\":\\"span\\",\\"id\\":\\"current-password-error\\",\\"key\\":\\"current-password-error-message\\",\\"className\\":\\"form-error\\",\\"message\\":\\"Oops! Let’s try that again — Invalid password.\\\\n       Please enter your password.\\"},{\\"id\\":\\"submit-button\\",\\"key\\":\\"zephr-submit-button\\",\\"type\\":\\"submit\\",\\"value\\":\\"Login\\"}]",
      "error": true,
      "errorCount": 1,
      "errors": Array [
        "invalid-password",
      ],
      "requireCaptcha": false,
    },
  },
  "headerHeight": 0,
  "loading": false,
  "player": Object {
    "currentTime": 0,
    "duration": 0,
    "loading": false,
    "playing": false,
    "seek": 0,
    "src": "",
    "visible": false,
    "volume": 0.5,
  },
  "route": Object {
    "action": "",
    "redirectStatus": 0,
    "redirectTo": false,
    "status": null,
  },
  "visible": Object {
    "megaMenu": false,
    "sliderAd": false,
    "sliderAdHasClosed": false,
  },
}
`;

exports[`Zephr Reducer Registration errors Should invalidate the email address field when an existing account email is entered 1`] = `
Object {
  "componentData": Object {},
  "components": Object {
    "defaults": Array [],
    "page": Object {},
    "providers": Object {},
  },
  "error": null,
  "forms": Object {
    "register": Object {
      "components": "[{\\"autoComplete\\":\\"username\\",\\"className\\":\\"zephr-input-email-address\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"email-address\\",\\"placeholder\\":\\"Enter your email address\\",\\"required\\":false,\\"type\\":\\"email\\",\\"invalid\\":true},{\\"type\\":\\"span\\",\\"id\\":\\"email-address-error\\",\\"key\\":\\"email-address-error-message\\",\\"className\\":\\"form-error\\",\\"message\\":\\"Oops! Let’s try that again — Account already exists!\\"},{\\"className\\":\\"zephr-input-full-name\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"full-name\\",\\"placeholder\\":\\"Enter your full name\\",\\"required\\":true,\\"type\\":\\"text\\"},{\\"autoComplete\\":\\"new-password\\",\\"className\\":\\"zephr-input-new-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"new-password\\",\\"placeholder\\":\\"Create a password for your account\\",\\"required\\":true,\\"type\\":\\"password\\"},{\\"autoComplete\\":\\"new-password\\",\\"className\\":\\"zephr-input-verify-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"verify-password\\",\\"placeholder\\":\\"Confirm your password\\",\\"required\\":true,\\"type\\":\\"password\\"},{\\"id\\":\\"terms-checkbox\\"},{\\"id\\":\\"submit-button\\",\\"key\\":\\"zephr-submit-button\\",\\"type\\":\\"submit\\",\\"value\\":\\"Create this account\\"}]",
      "error": true,
      "errors": Array [
        "user-already-exists",
      ],
    },
  },
  "headerHeight": 0,
  "loading": false,
  "player": Object {
    "currentTime": 0,
    "duration": 0,
    "loading": false,
    "playing": false,
    "seek": 0,
    "src": "",
    "visible": false,
    "volume": 0.5,
  },
  "route": Object {
    "action": "",
    "redirectStatus": 0,
    "redirectTo": false,
    "status": null,
  },
  "visible": Object {
    "megaMenu": false,
    "sliderAd": false,
    "sliderAdHasClosed": false,
  },
}
`;

exports[`Zephr Reducer Registration errors Should invalidate the email address field when an invalid email is entered 1`] = `
Object {
  "componentData": Object {},
  "components": Object {
    "defaults": Array [],
    "page": Object {},
    "providers": Object {},
  },
  "error": null,
  "forms": Object {
    "register": Object {
      "components": "[{\\"autoComplete\\":\\"username\\",\\"className\\":\\"zephr-input-email-address\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"email-address\\",\\"placeholder\\":\\"Enter your email address\\",\\"required\\":false,\\"type\\":\\"email\\",\\"invalid\\":true},{\\"type\\":\\"span\\",\\"id\\":\\"email-address-error\\",\\"key\\":\\"email-address-error-message\\",\\"className\\":\\"form-error\\",\\"message\\":\\"Oops! Let’s try that again — Please enter\\\\n       a valid email address.\\"},{\\"className\\":\\"zephr-input-full-name\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"full-name\\",\\"placeholder\\":\\"Enter your full name\\",\\"required\\":true,\\"type\\":\\"text\\"},{\\"autoComplete\\":\\"new-password\\",\\"className\\":\\"zephr-input-new-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"new-password\\",\\"placeholder\\":\\"Create a password for your account\\",\\"required\\":true,\\"type\\":\\"password\\"},{\\"autoComplete\\":\\"new-password\\",\\"className\\":\\"zephr-input-verify-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"verify-password\\",\\"placeholder\\":\\"Confirm your password\\",\\"required\\":true,\\"type\\":\\"password\\"},{\\"id\\":\\"terms-checkbox\\"},{\\"id\\":\\"submit-button\\",\\"key\\":\\"zephr-submit-button\\",\\"type\\":\\"submit\\",\\"value\\":\\"Create this account\\"}]",
      "error": true,
      "errors": Array [
        "email-address",
      ],
    },
  },
  "headerHeight": 0,
  "loading": false,
  "player": Object {
    "currentTime": 0,
    "duration": 0,
    "loading": false,
    "playing": false,
    "seek": 0,
    "src": "",
    "visible": false,
    "volume": 0.5,
  },
  "route": Object {
    "action": "",
    "redirectStatus": 0,
    "redirectTo": false,
    "status": null,
  },
  "visible": Object {
    "megaMenu": false,
    "sliderAd": false,
    "sliderAdHasClosed": false,
  },
}
`;

exports[`Zephr Reducer Registration errors Should invalidate the full name field when an invalid name is entered 1`] = `
Object {
  "componentData": Object {},
  "components": Object {
    "defaults": Array [],
    "page": Object {},
    "providers": Object {},
  },
  "error": null,
  "forms": Object {
    "register": Object {
      "components": "[{\\"autoComplete\\":\\"username\\",\\"className\\":\\"zephr-input-email-address\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"email-address\\",\\"placeholder\\":\\"Enter your email address\\",\\"required\\":false,\\"type\\":\\"email\\"},{\\"className\\":\\"zephr-input-full-name\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"full-name\\",\\"placeholder\\":\\"Enter your full name\\",\\"required\\":true,\\"type\\":\\"text\\",\\"invalid\\":true},{\\"type\\":\\"span\\",\\"id\\":\\"full-name-error\\",\\"key\\":\\"full-name-error-message\\",\\"className\\":\\"form-error\\",\\"message\\":\\"Oops! Let’s try that again — Please enter your full name.\\"},{\\"autoComplete\\":\\"new-password\\",\\"className\\":\\"zephr-input-new-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"new-password\\",\\"placeholder\\":\\"Create a password for your account\\",\\"required\\":true,\\"type\\":\\"password\\"},{\\"autoComplete\\":\\"new-password\\",\\"className\\":\\"zephr-input-verify-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"verify-password\\",\\"placeholder\\":\\"Confirm your password\\",\\"required\\":true,\\"type\\":\\"password\\"},{\\"id\\":\\"terms-checkbox\\"},{\\"id\\":\\"submit-button\\",\\"key\\":\\"zephr-submit-button\\",\\"type\\":\\"submit\\",\\"value\\":\\"Create this account\\"}]",
      "error": true,
      "errors": Array [
        "full-name",
      ],
    },
  },
  "headerHeight": 0,
  "loading": false,
  "player": Object {
    "currentTime": 0,
    "duration": 0,
    "loading": false,
    "playing": false,
    "seek": 0,
    "src": "",
    "visible": false,
    "volume": 0.5,
  },
  "route": Object {
    "action": "",
    "redirectStatus": 0,
    "redirectTo": false,
    "status": null,
  },
  "visible": Object {
    "megaMenu": false,
    "sliderAd": false,
    "sliderAdHasClosed": false,
  },
}
`;

exports[`Zephr Reducer Registration errors Should invalidate the password and verification fields when the passwords do not match 1`] = `
Object {
  "componentData": Object {},
  "components": Object {
    "defaults": Array [],
    "page": Object {},
    "providers": Object {},
  },
  "error": null,
  "forms": Object {
    "register": Object {
      "components": "[{\\"autoComplete\\":\\"username\\",\\"className\\":\\"zephr-input-email-address\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"email-address\\",\\"placeholder\\":\\"Enter your email address\\",\\"required\\":false,\\"type\\":\\"email\\"},{\\"className\\":\\"zephr-input-full-name\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"full-name\\",\\"placeholder\\":\\"Enter your full name\\",\\"required\\":true,\\"type\\":\\"text\\"},{\\"autoComplete\\":\\"new-password\\",\\"className\\":\\"zephr-input-new-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"new-password\\",\\"placeholder\\":\\"Create a password for your account\\",\\"required\\":true,\\"type\\":\\"password\\",\\"invalid\\":true},{\\"autoComplete\\":\\"new-password\\",\\"className\\":\\"zephr-input-verify-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"verify-password\\",\\"placeholder\\":\\"Confirm your password\\",\\"required\\":true,\\"type\\":\\"password\\",\\"invalid\\":true},{\\"type\\":\\"span\\",\\"id\\":\\"verify-password-error\\",\\"key\\":\\"verify-password-error-message\\",\\"className\\":\\"form-error\\",\\"message\\":\\"Oops! Let’s try that again — Passwords do not match.\\\\n       Please re-enter your password and try again.\\"},{\\"id\\":\\"terms-checkbox\\"},{\\"id\\":\\"submit-button\\",\\"key\\":\\"zephr-submit-button\\",\\"type\\":\\"submit\\",\\"value\\":\\"Create this account\\"}]",
      "error": true,
      "errorCount": null,
      "errors": Array [
        "verify-password",
      ],
      "type": "register",
    },
  },
  "headerHeight": 0,
  "loading": false,
  "player": Object {
    "currentTime": 0,
    "duration": 0,
    "loading": false,
    "playing": false,
    "seek": 0,
    "src": "",
    "visible": false,
    "volume": 0.5,
  },
  "route": Object {
    "action": "",
    "redirectStatus": 0,
    "redirectTo": false,
    "status": null,
  },
  "visible": Object {
    "megaMenu": false,
    "sliderAd": false,
    "sliderAdHasClosed": false,
  },
}
`;

exports[`Zephr Reducer Registration errors Should invalidate the password field when a weak password is entered 1`] = `
Object {
  "componentData": Object {},
  "components": Object {
    "defaults": Array [],
    "page": Object {},
    "providers": Object {},
  },
  "error": null,
  "forms": Object {
    "register": Object {
      "components": "[{\\"autoComplete\\":\\"username\\",\\"className\\":\\"zephr-input-email-address\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"email-address\\",\\"placeholder\\":\\"Enter your email address\\",\\"required\\":false,\\"type\\":\\"email\\"},{\\"className\\":\\"zephr-input-full-name\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"full-name\\",\\"placeholder\\":\\"Enter your full name\\",\\"required\\":true,\\"type\\":\\"text\\"},{\\"autoComplete\\":\\"new-password\\",\\"className\\":\\"zephr-input-new-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"new-password\\",\\"placeholder\\":\\"Create a password for your account\\",\\"required\\":true,\\"type\\":\\"password\\",\\"invalid\\":true},{\\"type\\":\\"span\\",\\"id\\":\\"new-password-error\\",\\"key\\":\\"new-password-error-message\\",\\"className\\":\\"form-error\\",\\"message\\":\\"Oops! Let’s try that again — Your password must be at\\\\n       least 8 characters, include one uppercase letter,\\\\n       one lowercase letter, and one symbol (e.g. !#$%&).\\"},{\\"autoComplete\\":\\"new-password\\",\\"className\\":\\"zephr-input-verify-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"verify-password\\",\\"placeholder\\":\\"Confirm your password\\",\\"required\\":true,\\"type\\":\\"password\\"},{\\"id\\":\\"terms-checkbox\\"},{\\"id\\":\\"submit-button\\",\\"key\\":\\"zephr-submit-button\\",\\"type\\":\\"submit\\",\\"value\\":\\"Create this account\\"}]",
      "error": true,
      "errors": Array [
        "password-not-strong",
      ],
    },
  },
  "headerHeight": 0,
  "loading": false,
  "player": Object {
    "currentTime": 0,
    "duration": 0,
    "loading": false,
    "playing": false,
    "seek": 0,
    "src": "",
    "visible": false,
    "volume": 0.5,
  },
  "route": Object {
    "action": "",
    "redirectStatus": 0,
    "redirectTo": false,
    "status": null,
  },
  "visible": Object {
    "megaMenu": false,
    "sliderAd": false,
    "sliderAdHasClosed": false,
  },
}
`;

exports[`Zephr Reducer Registration errors Should invalidate the terms of service checkbox if not checked 1`] = `
Object {
  "componentData": Object {},
  "components": Object {
    "defaults": Array [],
    "page": Object {},
    "providers": Object {},
  },
  "error": null,
  "forms": Object {
    "register": Object {
      "components": "[{\\"autoComplete\\":\\"username\\",\\"className\\":\\"zephr-input-email-address\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"email-address\\",\\"placeholder\\":\\"Enter your email address\\",\\"required\\":false,\\"type\\":\\"email\\"},{\\"className\\":\\"zephr-input-full-name\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"full-name\\",\\"placeholder\\":\\"Enter your full name\\",\\"required\\":true,\\"type\\":\\"text\\"},{\\"autoComplete\\":\\"new-password\\",\\"className\\":\\"zephr-input-new-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"new-password\\",\\"placeholder\\":\\"Create a password for your account\\",\\"required\\":true,\\"type\\":\\"password\\"},{\\"autoComplete\\":\\"new-password\\",\\"className\\":\\"zephr-input-verify-password\\",\\"defaultValue\\":\\"\\",\\"id\\":\\"verify-password\\",\\"placeholder\\":\\"Confirm your password\\",\\"required\\":true,\\"type\\":\\"password\\"},{\\"id\\":\\"terms-checkbox\\",\\"invalid\\":true},{\\"type\\":\\"span\\",\\"id\\":\\"terms-checkbox-error\\",\\"key\\":\\"terms-checkbox-error-message\\",\\"className\\":\\"form-error\\",\\"message\\":\\"Oops! Let’s try that again — You must agree to the terms\\\\n       of service in order to create an account.\\"},{\\"id\\":\\"submit-button\\",\\"key\\":\\"zephr-submit-button\\",\\"type\\":\\"submit\\",\\"value\\":\\"Create this account\\"}]",
      "error": true,
      "errors": Array [
        "terms-checkbox",
      ],
    },
  },
  "headerHeight": 0,
  "loading": false,
  "player": Object {
    "currentTime": 0,
    "duration": 0,
    "loading": false,
    "playing": false,
    "seek": 0,
    "src": "",
    "visible": false,
    "volume": 0.5,
  },
  "route": Object {
    "action": "",
    "redirectStatus": 0,
    "redirectTo": false,
    "status": null,
  },
  "visible": Object {
    "megaMenu": false,
    "sliderAd": false,
    "sliderAdHasClosed": false,
  },
}
`;
